File: backend/CoverMe.Backend.Core/Models/Process/DotCoverCliOptions.cs
using CoverMe.Backend.Core.Enums.Process;

namespace CoverMe.Backend.Core.Models.Process;

public class DotCoverCliOptions
{
    public required DotCoverCommand Command { get; init; }
    public required DotCoverReportType ReportType { get; init; }
    public required string OutputPath { get; init; }
    public required string ProjectFolderPath { get; init; }
    public bool HideAutoProperties { get; init; } = true;
    public bool NoBuild { get; init; } = true;
    public string CoverageFilter { get; set; } = string.Empty;
    public string TestsFilter { get; set; } = string.Empty;
}

File: src/src/rider/main/kotlin/com/jetbrains/rider/plugins/coverme/models/process/DotCoverCliOptions.kt
package com.jetbrains.rider.plugins.coverme.models.process

import com.jetbrains.rider.plugins.coverme.enums.process.DotCoverCliCommand
import com.jetbrains.rider.plugins.coverme.enums.process.DotCoverCliReportType

class DotCoverCliOptions(
    val command: DotCoverCliCommand,
    val reportType: DotCoverCliReportType,
    val outputPath: String,
    val projectFolderPath: String,
    val hideAutoProperties: Boolean,
    val noBuild: Boolean,
    val coverageFilter: String,
    val testsFilter: String
)